### PR TITLE
Connection pooling for flyteadmin db

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
1. Added new db option in config like ` maxOpenConnection,maxIdleConnection,connectionMaxLifetime `
2. Set db option for connection pooling  

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Added new config and set value in Database connection 
```
database:
  maxOpenConnection: 5
  maxIdleConnection: 25
  connectionMaxLifetime: 1000
```

## Tracking Issue
https://github.com/flyteorg/flyte/issues/881

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by katrogan — COMMENTED** [review]
It's a bit confusing but can you update https://github.com/flyteorg/flyteadmin/blob/master/pkg/runtime/interfaces/application_configuration.go#L6? The db config predates the runtime structure and contains several changes

Also were you able to test this locally?

File: pkg/runtime/interfaces/application_configuration.go [hunk 1]
```diff
@@ -20,6 +22,12 @@ type DbConfigSection struct {
 	ExtraOptions string `json:"options"`
 	// Whether or not to start the database connection with debug mode enabled.
 	Debug bool `json:"debug"`
+	// Max open connection in postgres
+	MaxOpenConnection     int           `json:"maxOpenConnection"`
```
Comment (katrogan) [inline pkg/runtime/interfaces/application_configuration.go:26]: you need to copy these over to the db configuration struct, right? https://github.com/flyteorg/flyteadmin/blob/a4abd4cbb0552a905b95b5d1e58e95baef9f7a7e/pkg/rpc/adminservice/base.go#L83 here and elsewhere

File: pkg/repositories/config/postgres.go [hunk 2]
```diff
@@ -2,21 +2,33 @@ package config
...
 	"github.com/flyteorg/flytestdlib/promutils"
 
 	"github.com/jinzhu/gorm"
 	_ "github.com/jinzhu/gorm/dialects/postgres" // Required to import database driver.
 )
 
-const Postgres = "postgres"
+const (
+	Postgres                     = "postgres"
+	DefaultMaxIdleConnection     = 25
```
Comment (katrogan) [inline pkg/repositories/config/postgres.go:15]: instead of setting these defaults here, can we set them in the runtime config when calling MustRegisterSection like so: https://github.com/flyteorg/flyteadmin/blob/master/pkg/runtime/application_config_provider.go#L25,L30

File: pkg/runtime/interfaces/application_configuration.go (hunk 1 above)
Comment (evalsocket) [reply pkg/runtime/interfaces/application_configuration.go:26]: @kanterov why do we need to pass these config in extraOption. I used gorm approach https://gorm.io/docs/generic_interface.html

File: pkg/repositories/config/postgres.go (hunk 2 above)
Comment (evalsocket) [reply pkg/repositories/config/postgres.go:15]: ok
